### PR TITLE
feat(ScrollGradient): new component ScrollGradient

### PR DESCRIPTION
**Pull request by jlongshore**
Contributes to [#4309](https://github.com/carbon-design-system/ibm-products/issues/4309)

Migrate **ScrollGradient** from `Products v1 / Security / ScrollGradient` to **Products v2** as `ScrollGradient`. 

`v1` References: [GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/ScrollGradient/ScrollGradient.js), [Storybook](https://v1-carbon-for-ibm-products.netlify.app/?path=/story/security-components-scrollgradient-legacy--vertical-scrolling).

#### What did you change?

- direction prop - removed. Component should manage both horizontal and vertical scrolls. 
- color prop - no longer required - defaults to layout-01

#### How did you test and verify your work?
- [x] Storybook review.
- [x] Tests migrated and added to.
- [x] Design review by Cameron Calder.

### CODE REVIEW

File: packages/ibm-products/src/components/ScrollGradient/ScrollGradient.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+          >
+            {children}
+          </div>
+        </div>
+
+        {/* Gradient elements */}
+        {!hideStartGradient && (
+          <>
+            <div
+              className={cx(`${blockClass}__start-vertical`)}
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ScrollGradient/ScrollGradient.js:135]: `cx` is unnecessary here

File: packages/ibm-products/src/components/ScrollGradient/ScrollGradient.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+              style={{
+                right: gradientRight,
+                backgroundImage: `linear-gradient(0deg, transparent, ${color} 90%)`,
+              }}
+              role="presentation"
+              aria-hidden
+            />
+            <div
+              className={cx(`${blockClass}__start-horizontal`)}
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ScrollGradient/ScrollGradient.js:144]: `cx` is unnecessary here

File: packages/ibm-products/src/components/ScrollGradient/ScrollGradient.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                backgroundImage: `linear-gradient(-90deg, transparent, ${color} 90%)`,
+                bottom: gradientBottom,
+              }}
+              role="presentation"
+              aria-hidden
+            />
+          </>
+        )}
+        <div
+          className={cx(`${blockClass}__end-vertical`)}
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ScrollGradient/ScrollGradient.js:155]: `cx` is unnecessary here

File: packages/ibm-products/src/components/ScrollGradient/ScrollGradient.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+          style={{
+            right: gradientRight,
+            bottom: gradientBottom,
+            backgroundImage: `linear-gradient(0deg, ${color} 10%, transparent)`,
+          }}
+          role="presentation"
+          aria-hidden
+        />
+        <div
+          className={cx(`${blockClass}__end-horizontal`)}
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ScrollGradient/ScrollGradient.js:165]: `cx` is unnecessary here

**Review by davidmenendez — CHANGES_REQUESTED** [review]
looks awesome. just 1 small thing

**Review by elycheea — COMMENTED** [review]
Also need to remove `story-structure.js` and follow the updated story format.
https://github.com/carbon-design-system/ibm-products/pull/4534

**Review by elycheea — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**